### PR TITLE
Cfg: Add support to reset admin password at startup (setting: `reset_admin_password_at_start`)

### LINKED ISSUE #49055: Changed Environment Variables do not get picked up in Grafana even though the `env` command has the updated values
**What happened**:

1. Bring a Grafana Docker container up using the repository mentioned below. Use `make build` to generated `docker-compose.yml` file and then `make run` to bring the container up. The `GF_SECURITY_ADMIN_USER`, `GF_SECURITY_ADMIN_PASSWORD` are set via the `conf/grafana/.env`. Initial Values are set to `admin` and `supersecretpass` respectively
2. Grafana works and one can login with the credentials value as mentioned above on http://localhost:3000
3. Change the `GF_SECURITY_ADMIN_PASSWORD` in `conf/grafana/.env` to `supersecretpass123` and bring the container down using `make clean`
4. Create the updated `docker-compose.yml` file using `make build` and bring the container up using `make run`
5. This should make the container available on the same URL, however when one tries to login with the updated `GF_SECURITY_ADMIN_PASSWORD`, the login does not work. Grafana logs me in only with the previous password set

Grafana_Env_Var_Docker_bug

**What you expected to happen**:

Assuming the container is brought down and brought back up again, the new environment variables should be used to let the user login into Grafana.

Since `GF_SECURITY_ADMIN_PASSWORD` is now changed and is also reflected within the container's environment, the value should be the actual password for admin login and should not rely on the older password previously set.

**How to reproduce it (as minimally and precisely as possible)**:

- Test Environment Reproduction Repository: [Repo on GitHub](https://github.com/shantanoo-desai/compose-v2-env-vars-problem)

The repository provides a very simple example of reproducing the bug with all the relevant documentation necessary. It does not require any other dependency except an updated Docker Engine to v20.10.14 with the compose CLI v2.5.0 which is now shipped within the environment.

**Anything else we need to know?**:
The only possible way to overcome this bug, is to actually purge the docker volumes which may contain other valuable data. It is rather unfeasible to purge a complete docker volume to reset the new Admin password.

**Environment**:
- Grafana version: 8.4.0, 8.4.5
- Data source type & version: Docker Image: `grafana/grafana-oss:8.4.5` from Docker Hub
- OS Grafana is installed on: Tested with Docker on GNU/Linux, Docker with WSL2 on Windows 10
- User OS & Browser: Mozilla Firefox, Google Chrome
- Grafana plugins: N/A
- Others: docker version 20.10.14 and docker compose version v2.5.0

Output of `docker info` (if relevant)

```bash
Client:
 Context:    default
 Debug Mode: false
 Plugins:
  buildx: Docker Buildx (Docker Inc., v0.8.2)
  compose: Docker Compose (Docker Inc., v2.5.0)
  sbom: View the packaged-based Software Bill Of Materials (SBOM) for an image (Anchore Inc., 0.6.0)
  scan: Docker Scan (Docker Inc., v0.17.0)

Server:
 Containers: 16
  Running: 1
  Paused: 0
  Stopped: 15
 Images: 8
 Server Version: 20.10.14
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Native Overlay Diff: true
  userxattr: false
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Cgroup Version: 1
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: io.containerd.runc.v2 io.containerd.runtime.v1.linux runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 3df54a852345ae127d1fa3092b95168e4a88e2f8
 runc version: v1.0.3-0-gf46b6ba
 init version: de40ad0
 Security Options:
  seccomp
   Profile: default
 Kernel Version: 5.10.16.3-microsoft-standard-WSL2
 Operating System: Docker Desktop
 OSType: linux
 Architecture: x86_64
 CPUs: 12
 Total Memory: 24.78GiB
 Name: docker-desktop
 ID: AM3A:EIDA:J3BS:FMN2:MZMR:3CBY:YHXN:US7Q:BIKM:7POA:WAGC:MRWX
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 HTTP Proxy: http.docker.internal:3128
 HTTPS Proxy: http.docker.internal:3128
 No Proxy: hubproxy.docker.internal
 Registry: https://index.docker.io/v1/
 Labels:
 Experimental: false
 Insecure Registries:
  hubproxy.docker.internal:5000
  127.0.0.0/8
 Live Restore Enabled: false
```

### PR DESCRIPTION
**Pull request by BobDu**
Fixes #49055
I can not reopen #55980 . Reopen a new PR.

**Special notes for your reviewer:**

Please check that:
- [x] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [x] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: docs/sources/setup-grafana/configure-grafana/_index.md
Comment (chri2547) [inline docs/sources/setup-grafana/configure-grafana/_index.md:613]:
```suggestion
> Available in Grafana v10.1+
```

File: docs/sources/setup-grafana/configure-grafana/_index.md
Comment (chri2547) [inline docs/sources/setup-grafana/configure-grafana/_index.md:615]:
```suggestion
Resets the Grafana Admin user and password every time the Grafana server starts. The default value is `false`.
```

File: docs/sources/setup-grafana/configure-grafana/_index.md
Comment (chri2547) [inline docs/sources/setup-grafana/configure-grafana/_index.md:609]:
```suggestion
> In Grafana v10.1+, the behavior depends on the `reset_admin_every_time_run` value. When `reset_admin_every_time_run`is `false`, the password is set once on first run.
```

**Review by chri2547 — COMMENTED** [review]
Added some suggestions to the docs update. Please review before you accept to ensure I didn't change the technical meaning of what you wrote. I'll let engineering weigh in on the code updates. Thank you very much for your contribution!

**Comment by BobDu** [comment]
@chri2547  Tks. I have modified as you suggested.

**Comment by BobDu** [comment]
Do you have any more suggestions for cfg option names?
Is there a better fit than `reset_admin_every_time_run `?
@chri2547

**Comment by chri2547** [comment]
> Do you have any more suggestions for cfg option names? Is there a better fit than `reset_admin_every_time_run `? @chri2547

Good question! I'll let engineering address your question. They might have a particular naming convention they'd like to use.

**Comment by mildwonkey** [comment]
@BobDu if you're looking for nitpicking, I would go with  `reset_admin_password_at_start` or even just `reset_admin_password` and note that the admin password will reset every time grafana starts up if that's true.

**Comment by BobDu** [comment]
modified use cfg opt name `reset_admin_password_at_start `

**Review by mildwonkey — DISMISSED** [review]
Thank you so much for getting back to this, @BobDu !  

This isn't quite working correctly - I'm writing up my review now, but wanted to get this in place first :)

File: pkg/services/sqlstore/user.go
Comment (mildwonkey) [inline pkg/services/sqlstore/user.go:161]: It's not safe to assume that the admin user ID is always 1 - it most likely is, especially in the use cases you had in mind when adding this configuration, but it could be used in an instance where the admin user ID is not 1. We will need to verify that user is the admin. 

You can compare this to the reset-admin-password CLI command, which also assumes `1` is the default admin user ID, but validates that by first getting the user and checking the is_admin field: https://github.com/grafana/grafana/blob/main/pkg/cmd/grafana-cli/commands/reset_password_command.go#L53

Also, take a look at the user service [UpdatePassword](https://github.com/grafana/grafana/blob/main/pkg/services/user/userimpl/store.go#L366) method; your Update command should look closer to that one.

----

Please let me know if you need more specific suggestions, or even if you want me to push a commit to your PR branch - I don't want to over-explain and I'm not in a rush, but I also don't want you to feel abandoned (again) :)

Comment (BobDu) [reply pkg/services/sqlstore/user.go:161]: It would be really nice if you could push commit to the branch.

I just hope we can have this feature soon

Comment (mildwonkey) [reply pkg/services/sqlstore/user.go:161]: Sure thing! Just make sure you've [Allowed edits from maintainers](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork) on your fork and I'll see what I can do 😁

**Comment by mildwonkey** [comment]
Note: the "Check for patch conflicts" failure is expected - it can't run off a fork - and is not an issue.

**Comment by mildwonkey** [comment]
(don't mind me repeatedly changing the issue title - it's relevant for the changelog, and this hard to categorize)

**Pull request closed without merging**
